### PR TITLE
Updates to DevRev Destination - Contacts (RevUsers) no longer require accounts and tags are properly applied

### PR DESCRIPTION
**Pull request by RtpChris**
This pull request fixes one major issue, and updates Contact creation to match our updated policies

1) In the past, it was required to have an account for a Contact (RevUser), as this is no longer required the code has been updated to support this.
2) Tags set in CreateRevUser were only getting set when the Contact was created, not when there was an existing Contact.  Now the tag is applied in both cases.

## Testing

_Include any additional information about the testing you have completed to
ensure your changes behave as expected. For a speedy review, please check
any of the tasks you completed below during your testing._

- [x] Added [unit tests](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing) for new functionality
- [x] Tested end-to-end using the [local server](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing)
- [ ] [Segmenters] Tested in the staging environment

### CODE REVIEW

File: packages/destination-actions/src/destinations/devrev/createRevUser/index.ts [hunk 1]
```diff
@@ -124,65 +123,70 @@ const action: ActionDefinition<Settings, Payload> = {
...
         display_name: name,
-        external_ref: email,
-        org_id: revOrgId
+        external_ref: email
+      }
+
+      let requestUrl = ''
+      //Search using domain if the email domain is not blacklisted
+      // Otherwise search the external refs
+      if (domain !== email) requestUrl = `${getBaseUrl(settings)}${devrevApiPaths.accountsList}?domains="${domain}"`
```
Comment (joe-ayoub-segment) [inline packages/destination-actions/src/destinations/devrev/createRevUser/index.ts:141]: hi @RtpChris , 
Why would domain ever be equal to an email address?

**Comment by joe-ayoub-segment** [comment]
hi @RtpChris thanks for raising this PR. 

Could you please run `yarn types` and commit the result?

Would you be able to provide evidence of testing to demonstrate that the new code works as expected and doesn't break the experience for customers who are already using the I please?

**Comment by RtpChris** [comment]
Hey @joe-ayoub-segment - I've updated the types.

I've also attached examples of the testing I've done

API Calls

Blacklisted Email Domain

Non-blacklisted email domain

While going through and documenting this, I discovered a corner case I originally did not include proper handling for this change (where the email has a domain that is not blacklisted and there is no existing account).  I'm adding that now and will add additional coverage for that case shortly.

**Comment by RtpChris** [comment]
Ok I've updated the code and updated the tests - as part of that I realized that I had removed checks in the tests that should have been there.  I've added those back and just left the one test that should have been affected by this change modified.  The other changes to the tests were typos/reordering checks to make sense.

**Comment by RtpChris** [comment]
Here's the screenshots of the testing:

1) New RevUser, existing account. We can see that the new RevUser was created under the existing account
Screenshot 2024-07-16 180308
Screenshot 2024-07-16 180407

2) New RevUser, New Account. We can see that since the account wasn't found and the domain wasn't blacklisted, we created the new account properly

Screenshot 2024-07-16 180437

Screenshot 2024-07-16 180510

**Comment by RtpChris** [comment]
Also, more as an FYI, when I ran `yarn types`  it generated a types file in `liveramp-audiences`; I didn't include that file here since it's not my destination but figured it would be good to know about.

**Review by joe-ayoub-segment — APPROVED** [review]

**Comment by joe-ayoub-segment** [comment]
hi @RtpChris PR deployed.

**Pull request merged**